consensus/polybft/checkpoint_manager.go

```diff
@@ -68,13 +68,53 @@ type checkpointManager struct {
 	logger hclog.Logger
 	// state boltDb instance
 	state *State
+	// eventGetter gets exit events (missed or current) from blocks
+	eventGetter *eventsGetter[*ExitEvent]
 }
 
 // newCheckpointManager creates a new instance of checkpointManager
 func newCheckpointManager(key ethgo.Key, checkpointOffset uint64,
 	checkpointManagerSC types.Address, txRelayer txrelayer.TxRelayer,
 	blockchain blockchainBackend, backend polybftBackend, logger hclog.Logger,
 	state *State) *checkpointManager {
+	retry := &eventsGetter[*ExitEvent]{
```

Comment:
Not sure if all this impl belongs to here...

Reply:
It's a setup of the `eventGetter`, so it should be there when creating a new `checkpointManager`. Maybe I can extract these functions to seperate functions, and not be all inline like this, for readability.

Reply:
Moved that parsing logic to a separate function for improved readability:
 https://github.com/0xPolygon/polygon-edge/pull/1652/commits/e2abe2656c00ffd57be8f4a353e2d251676ba9a7